addons/scopes/functions/fnc_inventoryCheck.sqf

```diff
@@ -17,28 +17,119 @@
 
 params ["_player"];
 
-private _adjustment = ACE_player getVariable QGVAR(Adjustment);
-if (isNil "_adjustment") then {
-    // [Windage, Elevation, Zero]
-    _adjustment = [[0, 0, 0], [0, 0, 0], [0, 0, 0]];
-    ACE_player setVariable [QGVAR(Adjustment), _adjustment];
-    [ACE_player, QGVAR(Adjustment), _adjustment, 0.5] call EFUNC(common,setVariablePublic);
-};
+private _adjustment = ACE_player getVariable [QGVAR(Adjustment), [[0, 0, 0], [0, 0, 0], [0, 0, 0]]];
+private _updateAdjustment = false;
 
-if (isNil QGVAR(Optics)) then {
-    GVAR(Optics) = ["", "", ""];
-};
 private _newOptics = [_player] call FUNC(getOptics);
+private _newGuns = [primaryWeapon _player, secondaryWeapon _player, handgunWeapon _player];
 
 {
     if (_newOptics select _forEachIndex != _x) then {
         // The optic for this weapon changed, set adjustment to zero
         if (!((_adjustment select _forEachIndex) isEqualTo [0, 0, 0])) then {
```

Comment:
> Not sure, but I think we should move this down to line 72

We never write to `_newGuns`, so I don't think that the early definition can cause any trouble.

> because we should reset zeroing on weapon change, if the old weapon was zeroable

Nice catch :+1:. It would probably be easiest if any weapon change would reset the scope adjustments.